drivers/platform/stm32/stm32_spi.c

```diff
@@ -205,30 +206,33 @@ int32_t stm32_spi_write_and_read(struct spi_desc *desc,
 				 uint8_t *data,
 				 uint16_t bytes_number)
 {
-	int32_t	ret;
+	uint8_t *tx = data;
+	uint8_t *rx = data;
 	struct stm32_spi_desc *sdesc;
-
+	struct stm32_gpio_desc *gdesc;
+	SPI_TypeDef * SPIx;
 	if (!desc || !desc->extra || !data)
 		return -EINVAL;
 
 	if (!bytes_number)
 		return 0;
 
 	sdesc = desc->extra;
-
-	ret = gpio_set_value(sdesc->chip_select, GPIO_LOW);
-	if (ret < 0)
-		goto error;
-
-	ret = HAL_SPI_TransmitReceive(&sdesc->hspi, data, data, bytes_number,
-				      HAL_MAX_DELAY);
-	if (ret != HAL_OK) {
-		if (ret == HAL_TIMEOUT)
-			ret = -ETIMEDOUT;
-		else
-			ret = -EIO;
+	gdesc = sdesc->chip_select->extra;
+	SPIx = sdesc->hspi.Instance;
+
+	gdesc->port->BSRR = BIT(sdesc->chip_select->number) << 16;
+	__HAL_SPI_ENABLE(&sdesc->hspi);
+	while(bytes_number--) {
+		while(!(SPIx->SR & SPI_SR_TXE))
+			;
```

Comment:
nitpick: why the extra line.

Reply:
for clarity,
```
while(!(SPIx->SR & SPI_SR_TXE));
// next line
```
when reading the above code, one could think // next line is under while (easy to visually miss the semicolon)